zap/src/main/java/org/zaproxy/zap/view/ContextIncludePanel.java

```diff
@@ -64,6 +75,56 @@ public static String getPanelName(Context context) {
         return getPanelName(context.getId());
     }
 
+    private File selectImportFile() {
+        JFileChooser chooser = new JFileChooser(Constant.getContextsDir());
+        File file = null;
+
+        int rc = chooser.showOpenDialog(View.getSingleton().getMainFrame());
+        if (rc == JFileChooser.APPROVE_OPTION) {
+            file = chooser.getSelectedFile();
+            if (file == null || !file.exists()) {
+                return null;
+            }
+            return file;
+        }
+        return null;
+    }
+
+    private void addImportButton() {
+        String IMPORT_BUTTON_LABEL = Constant.messages.getString("context.include.import.label");
+        JButton importButton = new JButton(IMPORT_BUTTON_LABEL);
+        importButton.addActionListener(
+                new ActionListener() {
+                    @Override
+                    public void actionPerformed(ActionEvent evt) {
+                        File file = selectImportFile();
+                        if (file != null) {
+                            List<String> regexes = regexesPanel.getRegexes();
+
+                            String line = null;
+                            try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+                                while ((line = br.readLine()) != null) {
+                                    String trimmedRegex = line.trim();
+                                    if (!regexes.contains(trimmedRegex)) {
+                                        Pattern.compile(trimmedRegex, Pattern.CASE_INSENSITIVE);
+                                        regexes.add(trimmedRegex);
+                                    }
+                                }
+                            } catch (IOException | PatternSyntaxException e) {
+                                View.getSingleton()
+                                        .showWarningDialog(
+                                                Constant.messages.getString(
+                                                        "context.include.import.error",
+                                                        e.getLocalizedMessage()));
+                                return;
+                            }
+                            regexesPanel.setRegexes(regexes);
+                        }
+                    }
+                });
```

Comment:
The ActionListener creation could be extracted to a new private method (`getImportActionListener` or something like that, create it if null, just return it if not). Then, the addition of the listener could be done with a lambda `importButton.addActionListener(e -> getImportActionListener());`.

Or something like that, hopefully that gives you enough to work with. Feel free to ask questions if I haven't been clear enough.

Reply:
While I am not a huge fan of lambdas I can certainly change the code to use it there. I do not fully get where there is a place for getImportActionListener when using a lambda for the ActionListener.
I would extract the actionPerformed method to a new method and then use this as lambda, so no ActionListener object needs to be created. As far as I understand that is the essence of using lambdas.
Could look like this:
```
public void importButtonClicked(ActionEvent event) {
    File file = selectImportFile();
    if (file != null) {
        List<String> regexes = regexesPanel.getRegexes();

        String line = null;
        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
            while ((line = br.readLine()) != null) {
                String trimmedRegex = line.trim();
                if (!regexes.contains(trimmedRegex)) {
                    Pattern.compile(trimmedRegex, Pattern.CASE_INSENSITIVE);
                    regexes.add(trimmedRegex);
                }
            }
        } catch (IOException | PatternSyntaxException e) {
            View.getSingleton()
                    .showWarningDialog(
                            Constant.messages.getString(
                                    "context.include.import.error",
                                    e.getLocalizedMessage()));
            return;
        }
        regexesPanel.setRegexes(regexes);
    }
}


private void addImportButton() {
    JButton importButton = new JButton(Constant.messages.getString("context.include.import.label"));
    importButton.addActionListener(e -> importButtonClicked(e));
    regexesPanel.addButton(importButton);
}
```

Reply:
Ya that’s what I was envisioning even if I wasn’t articulating it well 🤪

Reply:
change to lambda (and moved all changes to MultipleRegexesOptionsPanel as per other discussion)